src/main/java/org/apache/mesos/hdfs/state/LiveState.java

```diff
@@ -1,103 +1,123 @@
-  private final Set<Protos.TaskID> nameNodes;
-  private final Map<Protos.TaskID, String> taskSlaveMap;
-  private final Set<String> journalNodeHosts;
-  private final Set<String> nameNodeHosts;
-  private final HashSet<Protos.TaskID> stagingTasks;
+  private Set<Protos.TaskInfo> stagingTasks = new HashSet<>();
+  private AcquisitionPhase currentAcquisitionPhase = AcquisitionPhase.JOURNAL_NODES;
+  private LinkedHashMap<Protos.TaskID, Protos.TaskStatus> runningTasks = new LinkedHashMap<>();
 
-  public LiveState() {
-    taskHostMap = new HashMap<>();
-    taskSlaveMap = new HashMap<>();
-    journalNodes = new HashSet<>();
-    nameNodes = new HashSet<>();
-    journalNodeHosts = new HashSet<>();
-    nameNodeHosts = new HashSet<>();
-    stagingTasks = new HashSet<>();
+  public void addStagingTask(Protos.TaskInfo taskInfo) {
+    stagingTasks.add(taskInfo);
   }
-  
-  public Map<Protos.TaskID, String> getTaskHostMap() {
-    return taskHostMap;
+
+  public int getStagingTasksSize() {
+    return stagingTasks.size();
+  }
+
+  public void removeStagingTask(final Protos.TaskID taskID) {
+    Set<Protos.TaskInfo> toRemove = new HashSet<>();
+    for (Protos.TaskInfo taskInfo : stagingTasks ) {
+      if (taskInfo.getTaskId().equals(taskID)) {
+        toRemove.add(taskInfo);
+      }
+    }
+    stagingTasks.removeAll(toRemove);
+  }
+
+  public LinkedHashMap<Protos.TaskID, Protos.TaskStatus> getRunningTasks() {
+    return runningTasks;
   }
 
-  public Map<Protos.TaskID, String> getTaskSlaveMap() {
-    return taskSlaveMap;
+  public void removeTask(Protos.TaskID taskId) {
+    runningTasks.remove(taskId);
   }
 
-  public Set<Protos.TaskID> getJournalNodes() {
-    return journalNodes;
+  public void updateTaskForStatus(Protos.TaskStatus status) {
+    runningTasks.put(status.getTaskId(), status);
   }
 
-  public Set<Protos.TaskID> getNameNodes() {
-    return nameNodes;
+  public AcquisitionPhase getCurrentAcquisitionPhase() {
+    return currentAcquisitionPhase;
   }
 
-  public Set<String> getJournalNodeHosts() {
-    return journalNodeHosts;
+  public void transitionTo(AcquisitionPhase phase) {
+    this.currentAcquisitionPhase = phase;
   }
 
-  public Set<String> getNameNodeHosts() {
-    return nameNodeHosts;
+  public int getJournalNodeSize() {
+    return countOfRunningTasksWith(HDFSConstants.JOURNAL_NODE_ID);
   }
 
-  public Set<Protos.TaskID> getStagingTasks() {
-    return stagingTasks;
+  public int getNameNodeSize() {
+    return countOfRunningTasksWith(HDFSConstants.NAME_NODE_TASKID);
   }
 
-  public boolean notInDfsHosts(String slaveId) {
-    return !taskSlaveMap.values().contains(slaveId);
+  public Protos.TaskID getFirstNameNodeTaskId() {
+    if (getNameNodeSize() >= 1) {
+      return getNamenodeTaskIds().get(0);
+    } else {
+      return null;
+    }
+  }
+
+  public Protos.TaskID getSecondNameNodeTaskId() {
+    if (getNameNodeSize() == HDFSConstants.TOTAL_NAME_NODES) {
```

Comment:
We want this set as 2 since for HA mode we need 2 NN's and we are not supporting federated NN's. We don't want to duplicate the code everywhere so use of constants instead of magic numbers is better.